### PR TITLE
Fixes formatting and typos in documentation

### LINKED ISSUE #592: [BUG] Updates needed in README.md, CONTRIBUTING.md, etc.
**Describe the bug**
There are some documentation issues that need to be addressed in the repo docs.

- [x] CONTRIBUTING.md says we're in a pre-alpha state. 
- [x] CONTRIBUTING.md has a "Feature Requests" section, but doesn't link to the Feature Request Issue template.
- [x] CONTRIBUTING.md has “w00t!!!" at the very end.
        PR: https://github.com/opensearch-project/OpenSearch-Dashboards/pull/598
- [x] In /packages/osd-opensearch/README.md it references a trial/basic license  (@ananzh)
       PR: https://github.com/opensearch-project/OpenSearch-Dashboards/pull/602
- [x] The [github_checks_reporter.json](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/github_checks_reporter.json) file references Kibana.  (@ananzh)
       PR: https://github.com/opensearch-project/OpenSearch-Dashboards/pull/603
- [x] [package.json](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/package.json#L22) has a broken link to https://www.opensearch.org/products/kibana  (@ananzh)
       PR: https://github.com/opensearch-project/OpenSearch-Dashboards/pull/604
- [x] [/packages/opensearch-eslint-config-opensearch-dashboards/package.json](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/packages/opensearch-eslint-config-opensearch-dashboards/package.json) and README.md still references elastic and kibana (@ananzh)
       PR: https://github.com/opensearch-project/OpenSearch-Dashboards/pull/605
- [x] [/packages/opensearch-datemath/readme.md](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/packages/opensearch-datemath/readme.md) still references Kibana (@ananzh)
       PR: https://github.com/opensearch-project/OpenSearch-Dashboards/pull/606
- [x] [/packages/opensearch-safer-lodash-set/LICENSE](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/packages/opensearch-safer-lodash-set/LICENSE#L15) references the kibana repo (@ananzh)
       PR: https://github.com/opensearch-project/OpenSearch-Dashboards/pull/607
- [x] [/packages/osd-eslint-import-resolver-opensearch-dashboards/lib/get_project_root.js](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/packages/osd-eslint-import-resolver-opensearch-dashboards/lib/get_project_root.js#L43-L44) incorrectly references the @elastic/eslint-import-resolver-kibana package (twice)
  Comment: https://github.com/opensearch-project/OpenSearch-Dashboards/issues/592#issuecomment-876806735
  https://github.com/opensearch-project/OpenSearch-Dashboards/pull/1697
- [x] [/packages/osd-optimizer/README.md](https://github.com/opensearch-project/OpenSearch-Dashboards/blame/main/packages/osd-optimizer/README.md#L11) references a kibana GitHub Issue (@ananzh)
       PR: https://github.com/opensearch-project/OpenSearch-Dashboards/pull/608
       Comment: https://github.com/opensearch-project/OpenSearch-Dashboards/issues/592#issuecomment-877408373
- [x] [/packages/osd-plugin-generator/README.md](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/packages/osd-plugin-generator/README.md#compatibility) references "versions before OpenSearch Dashboards 6.3" which is a Kibana version. Also references the Kibana plugin template. (@ananzh)
       PR: https://github.com/opensearch-project/OpenSearch-Dashboards/pull/609
- [x] [/packages/osd-plugin-helpers/README.md](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/packages/osd-plugin-helpers/README.md#versions) references "versions before OpenSearch Dashboards 6.3" which is a Kibana version. Also references the Kibana plugin helpers repo. (@ananzh)
       PR: https://github.com/opensearch-project/OpenSearch-Dashboards/pull/610
- [x] [/packages/osd-pm/README.md](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/packages/osd-pm/README.md#why-osdpm) has an incorrect reference to elastic-eslint-config-kibana. (@ananzh)
       PR: https://github.com/opensearch-project/OpenSearch-Dashboards/pull/611    
- [x] [/src/core/PRINCIPLES.md](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/core/PRINCIPLES.md) references Kibana (@ananzh)
       PR: https://github.com/opensearch-project/OpenSearch-Dashboards/pull/599      
- [x] [/src/core/README.md](https://github.com/opensearch-project/OpenSearch-Dashboards/tree/main/src/core#plugin-development) has broken links: Core Public API, Core Server API, and OpenSearch Dashboards Platform Plugin API 
      Comment: https://github.com/opensearch-project/OpenSearch-Dashboards/issues/592#issuecomment-876689032
      Leaving for now.
- [x] [/src/core/TESTING.md](https://github.com/opensearch-project/OpenSearch-Dashboards/blame/main/src/core/TESTING.md#L275) references a closed Kibana PR (@ananzh)
       PR: https://github.com/opensearch-project/OpenSearch-Dashboards/pull/597
- [x] The Dashboards README.md is missing the Logo, Code of Conduct, License, and Copyright sections
- [x] The Dashboards README.md includes a "Running tests" section.
- [x] The Dashboards README.md has a "How you can help" section.
- [x] The Dashboards README.md includes "Guiding Principles" which is duplicated from the website.
- [x] MAINTAINERS.md does not include a definition or an explanation of responsibilities.
       PR: https://github.com/opensearch-project/OpenSearch-Dashboards/pull/600
- [x] In src/core/server/saved_objects/migrations/README.md there is some very strange formatting on the 4th bullet.
  https://github.com/opensearch-project/OpenSearch-Dashboards/pull/1697
- [x] OpenSearch-Dashboards/src/plugins/expressions/README.md links to an elastic style URL "See Canvas documentation" but is on opensearch.org
  PR: https://github.com/opensearch-project/OpenSearch-Dashboards/pull/1438 
- [x] NOTICE.txt has a different format from OpenSearch's [NOTICE.txt](https://github.com/opensearch-project/OpenSearch/blob/main/NOTICE.txt). It includes licenses from a handful of dependencies.
       Issue: https://github.com/opensearch-project/OpenSearch-Dashboards/issues/765
- [x] The Dashboards repo does not have a SECURITY.md file @kavilla 
- [x] TYPESCRIPT.md references a GitHub issue in the "[EUI component is missing types](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/TYPESCRIPT.md#eui-component-is-missing-types)" section for a list of types in EUI but this list is out of date.
  https://github.com/opensearch-project/OpenSearch-Dashboards/pull/1697
- [x] "Avoid any" is duplicated in [TYPESCRIPT.md](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/TYPESCRIPT.md#use-any-as-little-as-possible) and [DEVELOPER_GUIDE.md](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/DEVELOPER_GUIDE.md#avoid-any-whenever-possible).

In addition, there are some open questions around telemetry to validate before resolve this two issues;
- [x] In/packages/osd-telemetry-tools/GUIDELINE.md it references telemetry - didn't we remove that as part of the forking?
- [x] In OpenSearch-Dashboards/src/plugins/telemetry/README.md references telemetry and telemetry.opensearch.org. Didn't we remove this whole thing?
[Response] we removed the URLs to publishing any telemetry, but the general code is still there. We did not want to remove it in case we wanted to have a replacement for it.

### PR DESCRIPTION
**Pull request by tmarkley**
### Description
Fixes formatting and typos in documentation
 
### Issues Resolved
Resolves #592
 
### Check List
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Review by joshuarrrr — APPROVED** [review]
LGTM

File: src/core/server/saved_objects/migrations/README.md [hunk 1]
```diff
@@ -19,7 +19,7 @@ Here is the gist of what happens if an index migration is necessary:
   - Create an alias `.opensearch_dashboards` that points to `.opensearch_dashboards_1`
 - Create a `.opensearch_dashboards_2` index
 - Copy all documents from `.opensearch_dashboards_1` into `.opensearch_dashboards_2`, running them through any applicable migrations
-- # Point the `.opensearch_dashboards` alias to `.opensearch_dashboards_2`
+- Point the `.opensearch_dashboards` alias to `.opensearch_dashboards_2`
```
Comment (kavilla) [inline src/core/server/saved_objects/migrations/README.md:22]: Tbh these values are incorrect. Do we think we should update this to be the actual system index that we use?

Comment (joshuarrrr) [reply src/core/server/saved_objects/migrations/README.md:22]: We should, but I'd generally vote for opening a new issue and PR to do so.

Comment (tmarkley) [reply src/core/server/saved_objects/migrations/README.md:22]: Agreed, @kavilla can you open an issue to address the required changes there?

Comment (kavilla) [reply src/core/server/saved_objects/migrations/README.md:22]: Added: https://github.com/opensearch-project/OpenSearch-Dashboards/issues/1739

**Pull request merged**